server/src/graphql/mutations/linkDatabase.ts

```diff
@@ -0,0 +1,95 @@
+import { dbTypeToDokkuPlugin } from './../utils';
+import { MutationResolvers } from '../../generated/graphql';
+import { prisma } from '../../prisma';
+import { dokku } from '../../lib/dokku';
+import { sshConnect } from '../../lib/ssh';
+
+export const linkDatabase: MutationResolvers['linkDatabase'] = async (
+  _,
+  { input },
+  { userId }
+) => {
+  if (!userId) {
+    throw new Error('Unauthorized');
+  }
+
+  const { databaseId, appId } = input;
+
+  // We find database to link
+  const database = await prisma.database.findOne({
+    where: {
+      id: databaseId,
+    },
+  });
+
+  // We find app to link
+  const app = await prisma.app.findOne({
+    where: {
+      id: appId,
+    },
+  });
+
+  if (!app) {
+    throw new Error(`App with ID ${appId} not found`);
+  }
+
+  if (app.userId !== userId) {
+    throw new Error(`App with ID ${appId} does not belong to ${userId}`);
+  }
+
+  if (!database) {
+    throw new Error(`Database with ID ${databaseId} not found`);
+  }
+
+  if (database.userId !== userId) {
+    throw new Error(
+      `Database with ID ${databaseId} does not belong to ${userId}`
+    );
+  }
+
+  const dbType = dbTypeToDokkuPlugin(database.type);
+
+  const ssh = await sshConnect();
+
+  const isLinked = await dokku.database.linked(
```

Comment:
you can also check if the link exist with a db query, it might actually be faster :)

Reply:
@pradel nice will get rid of this, if we can just check that ad not via dokku, then it is amazing as it makes couple of things easier in my next PR.